### ISSUE TITLE
Crash when open after building on device

### ISSUE THREAD

**Issue by nehtnet** [fridakitten/FridaCodeManager#24]
When I build the app with “make all” command and after install the .deb and try to open it crashes with this error:

`Library Validation failed: Rejecting '/private/preboot/4DEACDE4F107B8B078BF324C04971CED94E60B4A899E9EB37443274392B7BE2780EFD1281E97BFCB08C788297CBDD5CA/dopamine-kqnqTB/procursus/usr/lib/llvm-16/lib/libclang.dylib' (Team ID: none, platform: no) for process 'swifty(33020)' (Team ID: T8ALTGMVXN, platform: no), reason: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.`

Any idea? Thanks

**Comment by neht-net** [comment]
Now i can build the app with “make all” command but I get this error after install the .deb and try to execute:

Library Validation failed: Rejecting '/private/preboot/4DEACDE4F107B8B078BF324C04971CED94E60B4A899E9EB37443274392B7BE2780EFD1281E97BFCB08C788297CBDD5CA/dopamine-kqnqTB/procursus/usr/lib/llvm-16/lib/libclang.dylib' (Team ID: none, platform: no) for process 'swifty(33020)' (Team ID: T8ALTGMVXN, platform: no), reason: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.

Any idea? Thanks

**Comment by neht-net** [comment]
Also the ipa generated with “make stock” crashes when opening (sideloaded on a stock iPhone).

**Comment by neht-net** [comment]
> When I build the app with “make all” command and after install the .deb and try to open it crashes with this error:
> 
> `Library Validation failed: Rejecting '/private/preboot/4DEACDE4F107B8B078BF324C04971CED94E60B4A899E9EB37443274392B7BE2780EFD1281E97BFCB08C788297CBDD5CA/dopamine-kqnqTB/procursus/usr/lib/llvm-16/lib/libclang.dylib' (Team ID: none, platform: no) for process 'swifty(33020)' (Team ID: T8ALTGMVXN, platform: no), reason: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.`
> 
> Any idea? Thanks

Just signing the file with “sudo ldid -S libclang.dylib“ made it work.

**Comment by fridakitten** [comment]
Yeah because procursus is stupid, back then they signed stuff like libclang

**Comment by neht-net** [comment]
> Also the ipa generated with “make stock” crashes when opening (sideloaded on a stock iPhone).

Solved with a couple of changes.

1. Makefile (line 40), add “linkfix” to the stock target:

`stock: greet compile_swift linkfix makechain_jailed ipa clean done`

2. Chainmaker/jailed.sh (line 8), add "libclang1-16" package:

`packages=("libiosexec1" "libedit0" "libzstd1" "libxar1" "ld64" "libtapi" "libllvm16" "libclang-cpp16" "libclang1-16" "clang-16" "libclang-common-16-dev" "libssl3" "libncursesw6")`

After these changes the app opens and works but now it crashes when trying to compile a project.
